index.js

```diff
@@ -88,6 +89,11 @@ Object.defineProperties(MediaSession.prototype, {
             }
             return [];
         }
+    },
+    channels: {
+        get: function () {
+            return this.pc._remoteDataChannels.concat(this.pc._localDataChannels);
```

Comment:
Sounds good to me. I'll update.
On Sun, Nov 22, 2015 at 1:13 PM Philipp Hancke <notifications@github.com>
wrote:

> In index.js
> <https://github.com/otalk/jingle-media-session/pull/15#discussion_r45562188>
> :
>
> > @@ -88,6 +89,11 @@ Object.defineProperties(MediaSession.prototype, {
> >              }
> >              return [];
> >          }
> > +    },
> > +    channels: {
> > +        get: function () {
> > +            return this.pc._remoteDataChannels.concat(this.pc._localDataChannels);
>
> oddly, there is no equivalent to getLocalStreams/getRemoteStreams.
> Managing the datachannel lifecycle is app reponsibility. I don't think the
> media session is the right place even though I agree we should be emitting
> channelAdded to enable the application (or any modules on top of this) to
> handle this.
>
> tl;dr: can we just do onAddChannel? ;-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/otalk/jingle-media-session/pull/15/files#r45562188>.
>